src/main/scala/io/adamnfish/pcb/Main.scala

```diff
@@ -215,4 +198,72 @@ object Main {
       } yield b :: acc
     }
   }
+
+  // Collects all results and all errors instead of failing fast
+  def collectAllResults[A, B](
+      la: List[A]
+  )(f: A => Either[String, B]): (List[String], List[B]) = {
+    val (errors, successes) = la.foldLeft((List.empty[String], List.empty[B])) { case ((errors, successes), a) =>
+      f(a) match {
+        case Left(error) => (errors :+ error, successes)
+        case Right(success) => (errors, successes :+ success)
+      }
+    }
+    (errors, successes)
+  }
+
+  def processFilenames(names: List[String]): (List[String], List[Filenames]) = {
```

Comment:
Is this function really providing any value? If we're testing both the `getFilenameWithDirectory` function and `collectAllResults` function, then the integration of these working components (which is all this does) is something we can integration test.